### ISSUE TITLE
how to format/preprocess a dataset?

### ISSUE THREAD

**Issue by goastler** [prosopo/captcha#295]

**Comment by goastler** [comment]
A client has a dataset with `n` categories and `m` images. What options are available in terms of labelling?

1. single provider with whole dataset
    - the slowest option (single provider worth of traffic)
    - easiest to begin with, no preprocessing required
1. multiple providers with whole dataset
    - quicker because getting multiple provider's worth of traffic)
    - requires running multiple providers
    - requires multiple stakes (1 per provider)
    - expedites labelling throughput only
1. multiple providers with chunked dataset
    - same benefits+cons of running multiple providers
    - chunking the dataset means labelling does not overlap, so somewhat more efficient labelling (e.g. given image abc.png. With multiple providers, that image may end up being over-labelled as providers are not aware of each other's labelling. Via chunking, you can make a single provider responsible for part of the dataset to avoid over-labelling)
    - this requires preprocessing to split up the dataset
1. multiple providers with 1v1 dataset
    - same benefits+cons of running multiple providers
    - require preprocessing of the dataset into 1v1 datasets. E.g. given classes a/b/c in a single dataset, it would be split into three datasets: a or not a, b or not b, c or not c. This would duplicate the data 3 times, so requires more space
1. multiple providers with 1v1 dataset and chunked dataset
    - methods combined together for better throughput of labelling, best of both worlds
 
tldr:
- running multiple providers nets more traffic. More traffic means more labelling
- it doesn't really matter how you format your data, but chunking across multiple providers can avoid over labelling

**Comment by forgetso** [comment]
closed as we're happy with external providers hosting the same dataset multiple times
